### ISSUE TITLE
`while_let_loop` doesn't take into account break-with-value

### ISSUE THREAD

**Issue by Vurich** [rust-lang-nursery/rust-clippy#1948]
Minimal reproduction:

```rust
let a = Some(10);
let foo = loop {
    if let Some(b) = a {
        break Some(a);
    } else {
        break None;
    }
};
```

Produces this error:

```
warning: this loop could be written as a `while let` loop
 --> src/main.rs:3:15
  |
3 |       let foo = loop {
  |  _______________^
4 | |         if let Some(b) = a {
5 | |             break Some(a);
6 | |         } else {
7 | |             break None;
8 | |         }
9 | |     };
  | |_____^ help: try: `while let Some(b) = a { .. }`
  |
  = note: #[warn(while_let_loop)] on by default
  = help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#while_let_loop
```

(Also a bunch of other errors because clippy is smart)

However, this loop can't be written this way since break-with-value is only supported on unconditional loops.

**Comment by oli-obk** [comment]
Just needs a test for value-passing breaks and continues

**Comment by oli-obk** [comment]
Has been fixed in #2023
